File: app/src/main/java/com/blogspot/android_czy_java/beautytips/detail/dialogs/NewCommentDialog.java
package com.blogspot.android_czy_java.beautytips.detail.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;

import com.blogspot.android_czy_java.beautytips.R;
import com.blogspot.android_czy_java.beautytips.detail.firebase.CommentsFirebaseHelper;
import com.blogspot.android_czy_java.beautytips.listView.view.dialogs.NicknamePickerDialog;

public class NewCommentDialog extends DialogFragment {

    private String comment;
    private String recipeId;

    private AddCommentListener mAddCommentListener;

    public interface AddCommentListener {
        void closeCommentsWindow();
    }


    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.DialogStyle);

        builder.setMessage(R.string.comment_dialog_msg);
        builder.setTitle(R.string.comment_dialog_title);

        builder.setPositiveButton(R.string.comment_dialog_positive_button,
                new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                CommentsFirebaseHelper.saveComment(recipeId, comment);
                mAddCommentListener.closeCommentsWindow();
            }
        });

        builder.setNegativeButton(R.string.comment_dialog_negative_button,
                new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                    }
                });

        return builder.create();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        try {
            mAddCommentListener = (AddCommentListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString() + " must implement NicknamePickerDialog" +
                    ".NicknamePickerDialogListener");
        }
    }

    public void setComment(String comment, String recipeId) {
        this.comment = comment;
        this.recipeId = recipeId;
    }
}

File: app/src/main/java/com/blogspot/android_czy_java/beautytips/view/detail/dialogs/NewCommentDialog.kt
package com.blogspot.android_czy_java.beautytips.view.detail.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.fragment.app.DialogFragment

import com.blogspot.android_czy_java.beautytips.R
import com.blogspot.android_czy_java.beautytips.view.detail.firebase.CommentsFirebaseHelper

class NewCommentDialog : DialogFragment() {

    private var comment = ""
    private var recipeId = ""
    private var commentsNum = ""

    private lateinit var mAddCommentListener: AddCommentListener

    interface AddCommentListener {
        fun onNewCommentAdded()
    }


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val builder = AlertDialog.Builder(activity, R.style.DialogStyle)

        builder.setTitle(R.string.comment_dialog_title)
        builder.setMessage(R.string.comment_dialog_msg)

        builder.setPositiveButton(R.string.comment_dialog_positive_button
        ) { _, _ ->
            CommentsFirebaseHelper(mAddCommentListener).saveComment(recipeId, comment, commentsNum)
        }

        builder.setNegativeButton(R.string.comment_dialog_negative_button
        ) { _, _ -> }

        return builder.create()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)

        try {
            mAddCommentListener = context as AddCommentListener
        } catch (e: ClassCastException) {
            throw ClassCastException(context.toString() + " must implement NewCommentDialog." +
                    "AddCommentListener")
        }

    }

    fun setComment(comment: String, recipeId: String, commentsNum: String) {
        this.comment = comment
        this.recipeId = recipeId
        this.commentsNum = commentsNum
    }
}
